connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/ExecutePlanResponseReattachableIterator.scala

```diff
@@ -52,7 +52,7 @@ class ExecutePlanResponseReattachableIterator(
     request: proto.ExecutePlanRequest,
     channel: ManagedChannel,
     retryPolicy: GrpcRetryHandler.RetryPolicy)
-    extends java.util.Iterator[proto.ExecutePlanResponse]
+    extends AutoConsumableIterator[proto.ExecutePlanResponse]
```

Comment:
Consuming the iterator like AutoConsumableIterator does involve waiting for the execution to complete server side, and receiving all results from it. It may not be the intention of the user to wait for it when they abandon / close iterator.
Closing the iterator here should involve first sending an Interrupt to the server, and then a ReleaseExecute request.